lib/board.rb

```diff
@@ -0,0 +1,85 @@
+class Board
+  attr_reader :cells
+
+  def initialize
+
+    cell1 = Cell.new("A1")
+    cell2 = Cell.new("A2")
+    cell3 = Cell.new("A3")
+    cell4 = Cell.new("A4")
+    cell5 = Cell.new("B1")
+    cell6 = Cell.new("B2")
+    cell7 = Cell.new("B3")
+    cell8 = Cell.new("B4")
+    cell9 = Cell.new("C1")
+    cell10 = Cell.new("C2")
+    cell11 = Cell.new("C3")
+    cell12 = Cell.new("C4")
+    cell13 = Cell.new("D1")
+    cell14 = Cell.new("D2")
+    cell15 = Cell.new("D3")
+    cell16 = Cell.new("D4")
+
+    @cells = {
+      "A1"=> cell1, "A2"=> cell2, "A3"=> cell3, "A4"=> cell4,
+      "B1"=> cell5, "B2"=> cell6, "B3"=> cell7, "B4"=> cell8,
+      "C1"=> cell9, "C2"=> cell10, "C3"=> cell11, "C4"=> cell12,
+      "D1"=> cell13, "D2"=> cell14, "D3"=> cell15, "D4"=> cell16
+    }
+
+    @board_numerical = { "A1"=> 1, "A2"=> 2, "A3"=> 3, "A4"=> 4,
+    "B1"=> 5, "B2"=> 6, "B3"=> 7, "B4"=> 8,
+    "C1"=> 9, "C2"=> 10, "C3"=> 11, "C4"=> 12,
+    "D1"=> 13, "D2"=> 14, "D3"=> 15, "D4"=> 16}
+
+  end
+
+  #check passed coordinate (key) exists in @cells (hash)
+  def valid_coordinate?(coordinate)
+    if @cells[coordinate] == nil
+      false
+    else
+      true
+    end
+  end
+
+  #checks the following:
+  #ship length == number of coordinates,
+  #if coorinates have same letters (A1, A2), they have sequential numbers
+  #if coordinates have sequential letters (A1, D1), they have the same numbers
+  def valid_placement?(ship, coordinates_array)
+    #tracks if previous coord iteration was "same" or "different"
+    situation = " "
+    #tracks if coordinate is valid or not
+    valid = true
+    #checks ship length vs num coordinates
+    if coordinates_array.count != ship.length
```

Comment:
.length runs faster than .count and still passes the tests for coordinates_array

Reply:
noted and fixed